pkg/local_object_storage/metabase/iterators.go

```diff
@@ -56,61 +58,100 @@ func (db *DB) IterateExpired(epoch uint64, h ExpiredObjectHandler) error {
 	})
 }
 
-func (db *DB) iterateExpired(tx *bbolt.Tx, epoch uint64, h ExpiredObjectHandler) error {
+func keyToEpochOID(k []byte, expStart []byte) (uint64, oid.ID) {
+	if !bytes.HasPrefix(k, expStart) || len(k) < len(expStart)+32+objectKeySize {
+		return math.MaxUint64, oid.ID{}
+	}
+	var (
+		f256 = k[len(expStart) : len(expStart)+32]
+		oidB = k[len(expStart)+32:]
+		id   oid.ID
+	)
+	for i := range len(f256) - 8 {
+		if f256[i] != 0 { // BE integer, too big.
+			return math.MaxUint64, oid.ID{}
+		}
+	}
+	var err = id.Decode(oidB)
+	if err != nil {
+		return math.MaxUint64, oid.ID{}
+	}
+
+	return binary.BigEndian.Uint64(f256[32-8:]), id
+}
+
+func (db *DB) iterateExpired(tx *bbolt.Tx, curEpoch uint64, h ExpiredObjectHandler) error {
+	var (
+		expStart  = make([]byte, 1+len(object.AttributeExpirationEpoch)+1+1)
+		typPrefix = make([]byte, 1+objectKeySize+len(object.FilterType)+1)
```

Comment:
```suggestion
		typPrefix = make([]byte, 1+oid.Size+len(object.FilterType)+1)
```

also one of the `1` also can be reused as `len` expression or as a const